### PR TITLE
Show correct artifact run node details on artifact side drawer panel

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
JIRA: https://issues.redhat.com/browse/RHOAIENG-7544

## Description

Updated the way we search for the artifact details for a specific node. Also, I removed the status for the artifact node because an artifact node shouldn't have status. (I checked that in KF UI)

Screenshot 2024-06-06 at 3 22 58 PM

## How Has This Been Tested?

1. Import the pipeline listed in the JIRA ticket
2. Create a run using that pipeline
3. Compare the artifact details with the KF UI to make sure the info is correct

## Test Impact

N/A, it's topology side panel details change, hard to write test.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
@DaoDaoNoCode I think Yan wants artifact nodes to be green if they exist and then grey otherwise

see this mock: https://www.figma.com/design/etbN0ccCcZR1qKCpJXQNAg/Pipeline-phase-2?node-id=1585-97434&t=BZENynAkmHo29x6Z-0

**Comment by DaoDaoNoCode** [comment]
@Gkrumbach07 Updated PR based on the mockup

File: frontend/src/concepts/topology/utils.ts [hunk 1]
```diff
@@ -47,7 +46,7 @@ export const createArtifactNode = (
   data: {
     pipelineTask,
     artifactType,
-    runStatus,
+    runStatus: pipelineTask.metadata ? RunStatus.Succeeded : RunStatus.Skipped,
```
Comment (Gkrumbach07) [inline frontend/src/concepts/topology/utils.ts:49]: not skipped but just no status

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]
works fine just one comment

**Comment by Gkrumbach07** [comment]
This looks good. we just need one more reviewer to merge

/lgtm

**Review by jenny-s51 — APPROVED** [review]
LGTM @DaoDaoNoCode , tested in UI and against KF - artifact panel renders data as expected.

**Review by Gkrumbach07 — APPROVED** [review]
/approve

**Pull request merged**